From 13d383536bddc3f75eb9cb13c74f8f9cda9eaae6 Mon Sep 17 00:00:00 2001
Message-Id: <RFC-patch-1.5-13d383536bd-20230317T042408Z-avarab@gmail.com>
In-Reply-To: <RFC-cover-0.5-00000000000-20230317T042408Z-avarab@gmail.com>
References: <pull.1463.v2.git.git.1678925506.gitgitgadget@gmail.com>
	<RFC-cover-0.5-00000000000-20230317T042408Z-avarab@gmail.com>
From: =?UTF-8?q?=C3=86var=20Arnfj=C3=B6r=C3=B0=20Bjarmason?=
 <avarab@gmail.com>
Date: Fri, 17 Mar 2023 03:14:15 +0100
Subject: [RFC PATCH 1/5] config.h: move up "struct key_value_info"
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Move the declaration of the "struct key_value_info" earlier in the
file, in a subsequent commit we'll need it shortly after this line.

Signed-off-by: Ævar Arnfjörð Bjarmason <avarab@gmail.com>
---
 config.h | 14 +++++++-------
 1 file changed, 7 insertions(+), 7 deletions(-)

diff --git a/config.h b/config.h
index 7606246531a..571b92d674a 100644
--- a/config.h
+++ b/config.h
@@ -110,6 +110,13 @@ struct config_options {
 	} error_action;
 };
 
+struct key_value_info {
+	const char *filename;
+	int linenr;
+	enum config_origin_type origin_type;
+	enum config_scope scope;
+};
+
 /**
  * A config callback function takes three parameters:
  *
@@ -612,13 +619,6 @@ int git_config_get_expiry(const char *key, const char **output);
 /* parse either "this many days" integer, or "5.days.ago" approxidate */
 int git_config_get_expiry_in_days(const char *key, timestamp_t *, timestamp_t now);
 
-struct key_value_info {
-	const char *filename;
-	int linenr;
-	enum config_origin_type origin_type;
-	enum config_scope scope;
-};
-
 /**
  * First prints the error message specified by the caller in `err` and then
  * dies printing the line number and the file name of the highest priority
-- 
2.40.0.rc1.1034.g5867a1b10c5

